### PR TITLE
Closes #1598 1598_fix_create_single_dose: fix 'create_single_dose_dataset()'

### LINKED ISSUE #1598: Bug:  create_single_dose_dataset expects mandatory values that documentation says are defaults, related to issue #1408
### What happened?

create_single_dose_dataset function documentation describes ASTDT, ASTDTM, AENDT and AENDTM as function default values but running with admiral version 0.9.0 ASTDT, ASTDTM, AENDT and AENDTM are now apparently mandatory variables.  I expected to define my own variable names (as below) in the function as done previously and create a ex_single dataframe to use in ad_adae.R.
  start_date = EXSTDT,
  start_datetime = EXSTDTM,
  end_date = EXENDT,
  end_datetime = EXENDTM

### Session Information

Error in assert_data_frame():
! Required variables ASTDT, ASTDTM, AENDT and AENDTM are missing.

### Reproducible Example

Several EX preprocessing steps are required before the final step where the ex_mod dataframe contains derived date and datetime variables
ex_single <-
  create_single_dose_dataset(
    ex_mod,
    dose_freq = EXDOSFRQ,
    start_date = EXSTDT,start,
    datetime = EXSTDTM,
    end_date = EXENDT,
    end_datetime = EXENDTM
    lookup_table = dose_freq_lookup,
    lookup_column = CDISC_VALUE
  ) %>%
  # Select only unique values   
  distinct(STUDYID,USUBJID,EXTRT,EXDOSE,EXDOSFRQ,DCUTDT,DTHDT,EXSTDT,EXENDT)

### PR DESCRIPTION
**Pull request by bundfussr**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by sadchla-codes — APPROVED** [review]

File: R/create_single_dose_dataset.R [hunk 1]
```diff
@@ -411,15 +416,38 @@ create_single_dose_dataset <- function(dataset,
   dataset_part_2 <- dataset %>%
     filter(!!dose_freq != "ONCE")
 
-  dataset_part_2 <- dataset_part_2 %>%
-    left_join(lookup, by = as.character(quo_get_expr(dose_freq))) %>%
-    mutate(dose_periods = case_when(
+  if (quo_is_null(start_datetime)) {
+    min_hour_cases <- rlang::exprs(FALSE ~ 0)
```
Comment (zdz2101) [inline R/create_single_dose_dataset.R:420]: Do we have guidance on when to use `pkg_name::` inside the function body? Should `exprs` just be included in the:
 `#' @importFrom rlang` portion of `R/admiral-package.R` ?

File: R/create_single_dose_dataset.R [hunk 2]
```diff
@@ -364,9 +363,15 @@ create_single_dose_dataset <- function(dataset,
...
+    condition <- paste(condition, "| is.na(", as_name(start_datetime), ")")
+  }
+  if (!quo_is_null(end_datetime)) {
+    condition <- paste(condition, "| is.na(", as_name(end_datetime), ")")
+  }
   na_check <- dataset %>%
-    filter(is.na(!!start_date) | is.na(!!end_date) |
-      is.na(!!start_datetime) | is.na(!!end_datetime)) %>%
+    filter(!!parse_expr(condition)) %>%
     select(!!start_date, !!end_date, !!start_datetime, !!end_datetime)
```
Comment (zdz2101) [inline R/create_single_dose_dataset.R:375]: This is more me not being as comfortable with quosures but by making the `_datetime` arguments optional, will these respective columns always exist/this line always run properly?

File: R/create_single_dose_dataset.R (hunk 1 above)
Comment (bundfussr) [reply R/create_single_dose_dataset.R]: We do not have a strict guidance.

I have added `exprs()` to `admiral-package.R`.

File: R/create_single_dose_dataset.R (hunk 2 above)
Comment (bundfussr) [reply R/create_single_dose_dataset.R:375]: `select()` ignores arguments which are `NULL`. For the default values the arguments resolve to `select(ASTDT, AENDT, NULL, NULL)` which selects `ASTDT` and `AENDT`.

**Comment by bundfussr** [comment]
@sadchla-codes , @zdz2101 , could you (re-)approve the PR to trigger the "Check Templates" workflow?

**Review by sadchla-codes — APPROVED** [review]

**Comment by sadchla-codes** [comment]
@bundfussr  sure

**Pull request merged**